packs/mmonit/sensors/events_sensor.yaml

```diff
@@ -0,0 +1,22 @@
+---
+  class_name: "MmonitEventsSensor"
+  entry_point: "events_sensor.py"
+  description: "Sensor which monitors mmonit for events"
+  poll_interval: 10
+  trigger_types:
+    -
+      name: "new_event"
+      description: "Trigger which indicates that a new event has arrived at mmonit"
+      payload_schema:
+        type: "object"
+        properties:
+          host:
+            type: "string"
+          id:
+            type: "integer"
+          event:
```

Comment:
I presume that event['event'] has a defined set of values. Each of those could be defined as a separate TriggerType.

say event['event'] values are one of ['event_x', 'event_y', 'event_z']. Then you can map these to 3 triggertypes. So instead of 'new_event' the TriggerTypes would be 'event_x', 'event_y' and 'event_z'.

In a rule instead of referring to 'new_event' in the trigger field you can now refer to the 3 other types.

Again this is all dependent on if the values are know, the values are a small enough number and there is a requirement to perform different work on each of these events. It will provide some more visibility to a rule author.